File: app/src/main/java/com/myhexaville/restwithdatabinding/MainActivity.java
package com.myhexaville.restwithdatabinding;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import com.myhexaville.restwithdatabinding.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    private ActivityMainBinding mBinding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mBinding = DataBindingUtil.setContentView(this, R.layout.activity_main);

//        setupToolbar();

        setupPager();
    }

//    private void setupToolbar() {
//        setSupportActionBar(mBinding.toolbar);
//    }

    private void setupPager() {
        mBinding.pager.setAdapter(new PagerAdapter(getSupportFragmentManager()));
        mBinding.tabs.setupWithViewPager(mBinding.pager);
    }
}

File: app/src/main/java/com/myhexaville/restwithdatabinding/MainActivity.kt
package com.myhexaville.restwithdatabinding

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.util.Log
import com.myhexaville.restwithdatabinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private val LOG_TAG = "MainActivity"

    private var binding: ActivityMainBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView<ActivityMainBinding>(this, R.layout.activity_main)

        setupPager()
    }

    private fun setupPager() {
        val sortType = getSortType()

        binding!!.pager.adapter = PagerAdapter(supportFragmentManager, sortType)
        binding!!.tabs.setupWithViewPager(binding!!.pager)
    }

    private fun getSortType(): String {
        return if (intent.extras != null && intent.extras.containsKey("sort")) {
            intent.extras.getString("sort")
        } else {
            "popular"
        }
    }
}
